diffkemp/diffkemp.py

```diff
@@ -517,6 +524,9 @@ def text_indent(text, width):
         """Indent each line in the text by a number of spaces given by width"""
         return ''.join(" " * width + line for line in text.splitlines(True))
 
+    old_dir_abs_path = os.path.join(os.path.abspath(snapshot_dir_old), "")
+    new_dir_abs_path = os.path.join(os.path.abspath(snapshot_dir_new), "")
```

Comment:
Same as above - are the `join` calls necessary?

Reply:
I looked into that and it looks like paths of functions and macros are saved as absolute paths and paths of types are saved as relative paths. That is the reason why the function `replace` is used in call stacks (to get always a relative path even if the path is already relative).  
The `join` is "necessary" for adding `/` to the end of the snapshot path  (without that using `replace` on calls with abs path would return "relative path" starting with backslash `/`). I guess it could be done differently (like just adding a backslash to the snapshot path) but I used `join` because it was used like that before.

Reply:
Makes sense, thanks.